consensus/polybft/signer/signature.go

```diff
@@ -69,43 +56,36 @@ func UnmarshalSignature(raw []byte) (*Signature, error) {
 		return nil, errors.New("cannot unmarshal signature from empty slice")
 	}
 
-	p, err := bn254.NewG1().FromBytes(raw)
-	if err != nil {
+	g1 := new(bn256.G1)
+	if _, err := g1.Unmarshal(raw); err != nil {
 		return nil, err
 	}
 
-	return &Signature{p: p}, nil
-}
-
-// ToBigInt marshalls signature (which is point) to 2 big ints - one for each coordinate
-func (s Signature) ToBigInt() ([2]*big.Int, error) {
-	sig, err := s.Marshal()
-	if err != nil {
-		return [2]*big.Int{}, err
+	// check if it is the point at infinity
+	if g1.IsInfinity() {
+		return nil, errInfinityPoint
 	}
 
-	res := [2]*big.Int{
-		new(big.Int).SetBytes(sig[0:32]),
-		new(big.Int).SetBytes(sig[32:64]),
+	// check if not part of the subgroup
+	if !g1.InCorrectSubgroup() {
+		return nil, fmt.Errorf("incorrect subgroup")
 	}
 
-	return res, nil
+	return &Signature{g1: g1}, nil
 }
 
 // Signatures is a slice of signatures
 type Signatures []*Signature
 
-// Aggregate sums the given array of signatures
-func (s Signatures) Aggregate() *Signature {
-	g, newp := bn254.NewG1(), new(bn254.PointG1)
-
-	newp.Set(g.Zero())
+// Aggregate aggregates all signatures into one
+func (sigs Signatures) Aggregate() *Signature {
+	g1 := new(bn256.G1)
 
-	for _, x := range s {
-		if x.p != nil {
-			g.Add(newp, newp, x.p)
+	for _, sig := range sigs {
+		if sig.g1 != nil {
```

Comment:
Same as in the PublicKey aggregation.

Reply:
fixed https://github.com/0xPolygon/polygon-edge/pull/1161/commits/c39a660fed18b2128cee9bc0329b13d278d6c8b5